ui/src/parts/lightbox.tsx

```diff
@@ -0,0 +1,229 @@
+        <Fluent.ActionButton
+          styles={iconStyles}
+          onClick={() => setActiveImageIdx(activeImageIdx === images.length - 1 ? 0 : activeImageIdx + 1)}
+          iconProps={{ iconName: 'ChevronRight', style: { fontSize: '22px' } }}
+        />
+      </div>
+    </>,
+  CloseButton = ({ onClose }: { onClose: () => void }) =>
+    <Fluent.ActionButton
+      styles={iconStyles}
+      onClick={onClose}
+      iconProps={{ iconName: 'Cancel', style: { fontSize: '22px' } }}
+    />
+
+export const Lightbox = ({ visible, onDismiss, images, defaultImageIdx }: LightboxProps) => {
+  const
+    [activeImageIdx, _setActiveImageIdx] = React.useState(defaultImageIdx || 0),
+    { title, description } = images[activeImageIdx],
+    isGallery = images.length > 1,
+    src = getImageSrc(images[activeImageIdx]),
+    imageNavRef = React.useRef<HTMLDivElement | null>(null),
+    navImgRefs: React.RefObject<HTMLImageElement>[] = images.map(() => React.createRef()),
+    activeImageIdxRef = React.useRef(activeImageIdx),
+    setActiveImageIdx = (idx: U) => {
+      // Store activeImageIdx inside ref to make it accessible from window key event closure.
+      _setActiveImageIdx(idx)
+      activeImageIdxRef.current = idx
+    },
+    onClose = () => {
+      onDismiss()
+      setActiveImageIdx(defaultImageIdx || 0)
+      if (imageNavRef.current) imageNavRef.current.scrollLeft = 0
+    },
+    handleKeyDown = (ev: KeyboardEvent) => {
+      if (ev.keyCode === keys.esc) onClose()
+      const imgIdx = activeImageIdxRef.current
+      if (ev.keyCode === keys.right) setActiveImageIdx(imgIdx === images.length - 1 ? 0 : imgIdx + 1)
+      else if (ev.keyCode === keys.left) setActiveImageIdx(imgIdx === 0 ? images.length - 1 : imgIdx - 1)
+    },
+    imageNav = isGallery ? images.map((image, idx) =>
+      <div key={idx} className={css.navImgPlaceholder}>
+        <img
+          className={clas(css.img, css.navImg, 'lazy')}
+          ref={navImgRefs[idx]}
+          style={activeImageIdx === idx ? { filter: 'unset', border: '2px solid red' } : undefined}
+          alt={title}
+          data-src={getImageSrc(image)}
+          onClick={() => setActiveImageIdx(idx)}
+        />
+      </div>
+    ) : undefined
+
+  React.useEffect(() => {
+    // Set initial scroll position.
+    if (isGallery) navImgRefs[activeImageIdx].current!.scrollIntoView({ behavior: 'auto', inline: 'center', block: 'center' })
+    // Add keyboard events listener.
+    if (visible) window.addEventListener("keydown", handleKeyDown)
+    else window.removeEventListener("keydown", handleKeyDown)
+    return () => window.removeEventListener("keydown", handleKeyDown)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [visible])
+
+  // Handle image navigation scroll.
+  React.useEffect(() => {
+    if (isGallery) navImgRefs[activeImageIdx].current!.scrollIntoView({
+      behavior: (activeImageIdx === 0 || activeImageIdx === images.length - 1) ? 'auto' : 'smooth',
```

Comment:
Yes, everything what's implemented is tested in recent versions of Chrome, Firefox and Safari and it works.

Reply:
![image](https://user-images.githubusercontent.com/64769322/194807170-427c2651-4277-4d70-a814-8d0b87e64cfa.png)

The smooth behavior is only supported in Safari version from last month which is not enough. Any alternatives?

Reply:
The most modern feature we use without polyfills is CSS variables, which are supported ~ 5 years back across all browsers so it's good to comply with that in general. However, since this particular feature is about the `smooth` scroll which is more like a nice-to-have rather than a hard requirement, your `scrollBy` support matrix is fine as long as the main feature (the element shall be scrolled into view) will work even if snappy instead of smooth.